### PR TITLE
Store file names and sizes in partition parameters

### PR DESCRIPTION
**Pull request by NikhilCollooru**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java [hunk 1]
```diff
@@ -17,21 +17,43 @@
...
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static io.airlift.slice.Slices.utf8Slice;
 import static java.lang.String.format;
 
 public class HiveManifestUtils
 {
     private static final int FILE_SIZE_CHANNEL = 0;
     private static final int ROW_COUNT_CHANNEL = 1;
+    private static final String CHARSET_NAME = "ISO-8859-1";
+    private static final int COMPRESSION_LEVEL = 3; // default level
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java:50]: make it 7 maybe

**Review by highker — COMMENTED** [review]
"Add getFileSize() to retrieve the size of the file" LGTM

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java [hunk 2]
```diff
@@ -85,4 +107,122 @@ public static long getFileSize(Page statisticsPage, int position)
...
+                break;
+            }
+            start++;
+        }
+
+        if (isContinuousSequence) {
+            return fileNames.get(fileNames.size() - 1);
+        }
+
+        return compressFileNamesUsingRoaringBitmap(fileNames);
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java:153]: I think can blindly compressing with roaring bitmap. It compress continuous range very well.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java [hunk 3]
```diff
@@ -85,4 +107,122 @@ public static long getFileSize(Page statisticsPage, int position)
...
+    }
+
+    static List<String> decompressFileNames(String compressedFileNames)
+    {
+        // Check if the compressed fileNames string is a number
+        if (compressedFileNames.matches("\\d+")) {
+            long end = Long.valueOf(compressedFileNames);
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java:160]: `parseLong`

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java [hunk 4]
```diff
@@ -85,4 +107,122 @@ public static long getFileSize(Page statisticsPage, int position)
...
+
+            if (end == 0) {
+                return Collections.singletonList("0");
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java:163]: `ImmutableList.of("0")`?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java [hunk 5]
```diff
@@ -85,4 +107,122 @@ public static long getFileSize(Page statisticsPage, int position)
...
+            }
+
+            ImmutableList.Builder<String> fileNames = ImmutableList.builder();
+            for (long i = 0; i <= end; i++) {
+                fileNames.add(String.valueOf(i));
+            }
+
+            return fileNames.build();
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java:171]:
```java
return LongStream.range(0, end + 1).mapToObj(String::valueOf).collect(toImmutableList());

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveManifestUtils.java [hunk 6]
```diff
@@ -76,6 +82,45 @@ public void testCreatePartitionManifest()
...
+        for (int i = 0; i <= 100; i++) {
+            if (i % 10 == 0) {
+                continue;
+            }
+            fileNames.add(String.valueOf(i));
+        }
+
+        String compressedNonSequentialFileNames = compressFileNames(fileNames);
+        assertEquals(fileNames, decompressFileNames(compressedNonSequentialFileNames));
+    }
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveManifestUtils.java:109]: Maybe also add one after this by randomly generating a sequence of numbers. Same like the one you did in `testFileSizesCompression`

**Review by highker — APPROVED** [review]

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java (hunk 2 above)
Comment (NikhilCollooru) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java:153]: Not really. For a list of continuous 100 numbers, RoaringBitMap is taking 218 bytes. But where as "100" takes 3 bytes. So the current code is more memory efficient.

**Pull request merged**